### PR TITLE
fix(deps): update to angular 17.1.0

### LINKED ISSUE #2142: [Bug]: SBB Angular not working with Angular 17.1.0
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/sbb-angular/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/sbb-angular/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/orgs/sbb-design-systems/projects/1) for an issue that matches the one I want to file, without success.

### Bug type

Functionality

### Which version are you using

17.3.0

### Operating system

Any

### Browser / Browser version

Any

### Input mode

None

### Additional settings

_No response_

### What happened?

We recently upgraded our angular dependencies to the currently newest version, 17.1.0.
After this upgrade, the sbb-sidebar component does not automatically expand the expansion-panels inside, even when they have the "expanded" property set. Also the "+"/"-" Icon do not load anymore unless you click on the expansion-panel:
image

In the above picture, I didn't touch any of the expansion-panels except for clicking on Test2.

Also in the Documentation for the sbb-sidebar, the first StackBlitz example is broken and doesn't compile: https://angular.app.sbb.ch/angular/components/sidebar/examples

### Steps to reproduce the issue

1. Upgrade project to newest Angular version and add @sbb-esta/angular to the project
2. Create a sidebar component and add some items -> as per the example in the documentation

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mhaertwig**
This fixes some issues that occurred during the update to Angular 17.1.0.

Closes #2142

### CODE REVIEW

File: tools/dgeni/common/property-bindings.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ function getBindingPropertyData(
 ) {
   if (metadata) {
     const metadataValues: string[] = metadata.get(propertyName) || [];
-    const foundValue = metadataValues.find((value) => value.split(':')[0] === doc.name);
+    const foundValue = metadataValues.find((value) => value?.split(':')[0] === doc.name);
```
Comment (mhaertwig) [inline tools/dgeni/common/property-bindings.ts:44]: Necessary to allow defining input properties as objects.

File: .npmrc [hunk 2]
```diff
@@ -46,5 +46,6 @@ registry=https://registry.yarnpkg.com
 
 
 
+
```
Comment (jeripeierSBB) [inline .npmrc:49]: interesting file. 48 new lines :D

File: src/angular/table/table/table.ts [hunk 3]
```diff
@@ -57,12 +57,32 @@ export class SbbRecycleRows {}
...
+        It gets removed on the client so it doesn't affect the layout.
+      -->
+    @if (_isServer) {
+      <ng-content />
+    }
+    @if (_isNativeHtmlTable) {
+      <thead role="rowgroup">
+        <ng-container headerRowOutlet />
+      </thead>
+      <tbody class="mdc-data-table__content" role="rowgroup">
```
Comment (jeripeierSBB) [inline src/angular/table/table/table.ts:73]: does this css class has any effect?

File: src/angular/table/table/table.ts [hunk 4]
```diff
@@ -57,12 +57,32 @@ export class SbbRecycleRows {}
...
+    <ng-content select="colgroup, col" />
+    <!--
+        Unprojected content throws a hydration error so we need this to capture it.
+        It gets removed on the client so it doesn't affect the layout.
+      -->
+    @if (_isServer) {
+      <ng-content />
+    }
+    @if (_isNativeHtmlTable) {
+      <thead role="rowgroup">
```
Comment (jeripeierSBB) [inline src/angular/table/table/table.ts:70]: Are roles on thead, tbody and tfoot really needed? I thought they have implicit roles

**Review by jeripeierSBB — APPROVED** [review]
LGTM, a few questions. Thanks for fixing it!

File: src/angular/table/table/table.ts (hunk 3 above)
Comment (mhaertwig) [reply src/angular/table/table/table.ts]: Thanks, I copied this class unintentionally and removed it again.

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Thank you very much!

File: src/angular/table/table/table.ts (hunk 4 above)
Comment (mhaertwig) [reply src/angular/table/table/table.ts:70]: I also copied that but after reading https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/rowgroup_role, I agree with you: It does not make sense to add this extra role.

Comment (mhaertwig) [reply src/angular/table/table/table.ts:70]: After discussion: I keep the roles (just in case ;-) )

**Pull request merged**